File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& ht) {
        int i = 0;
        int n = ht.size();
        int j = n-1;

        int maxArea = 0;


        while(i<j){
            int reHt = min(ht[i],ht[j]);
            int wtt  = j-i;
            int resAr = reHt*wtt;
            maxArea = max(maxArea,resAr);

            if(ht[i]>ht[j]){
                j--;
            }
            else{
                i++;
            }

        }
        return maxArea;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length-1;

        int maxWtr = -1;
        
        while(i<j){
            int ht = Math.min(height[i],height[j]);
            int br = j-i;

            int currAr = ht*br;

            maxWtr = Math.max(maxWtr,currAr);

            if(height[i]<height[j]){
                i++;
            }
            else{
                j--;
            }
        }
        return maxWtr;
    }
}